docs/guide/env-and-mode.md

```diff
@@ -60,9 +60,11 @@ By default, Vite provides type definition for `import.meta.env`. While you can d
 To achieve, you can create an `env.d.ts` in `src` directory, then augment `ImportMetaEnv` like this:
 
 ```typescript
-interface ImportMetaEnv {
-  VITE_APP_TITLE: string
-  // more env variables...
+interface ImportMeta {
+  env: {
+    VITE_APP_TITLE: string
+    // more env variables...
+  }
 }
```

Comment:
How about this:

```suggestion
interface ImportMetaEnv extends Readonly<Record<string, string>> {
  readonly VITE_APP_TITLE: string
  // more env variables...
}

interface ImportMeta {
  readonly env: ImportMetaEnv
}
```

Reply:
well, in essence, it's the same, it's the dev's decision. 
Using the current docs' suggestion results in the project not compiling.

Reply:
The thing is that you completely override the env property and loses everything from `ImportMetaEnv`  
My suggested solution (_already working that way, in my projects_) only extends the already existing interfaces but adds project known properties

Beside that, env properties should be `readonly` due to you should be not allowed to change them at runtime